### PR TITLE
Fix ./mach test-tidy --faster issue

### PR DESCRIPTION
**Pull request by zakorgy**
issue #9778

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10074)

### CODE REVIEW

**Comment by zakorgy** [comment]
Unfortunately my previous fork has been deleted, and I had to create a new PR, here is the previous PR https://github.com/servo/servo/pull/9925

File: python/tidy.py [hunk 1]
```diff
@@ -613,11 +631,11 @@ def scan(faster=False, progress=True):
     checking_functions = (check_flake8, check_lock, check_webidl_spec, check_json)
     line_checking_functions = (check_license, check_by_line, check_toml, check_rust, check_spec)
     errors = collect_errors_for_files(files_to_check, checking_functions, line_checking_functions)
-
-    # wpt lint checks
-    wpt_lint_errors = check_wpt_lint_errors(get_wpt_files(faster, progress))
-    # collect errors
-    errors = itertools.chain(errors, wpt_lint_errors)
+    if not faster:
```
Comment (Wafflespeanut) [inline python/tidy.py]: I don't understand the reason behind this change. This won't run if `faster` is `True`, and the commit author has made changes to WPT files.

Comment (zakorgy) [inline python/tidy.py]: I got confused by the --help message. It says the following:
Command Arguments:
  --faster       Only check changed files and skip the WPT lint in tidy
  --no-progress  Don't show progress for tidy
I read the conversation in #9778 again, may I change the help message?

File: python/servo/testing_commands.py [hunk 2]
```diff
@@ -263,7 +263,8 @@ def test_content(self):
              description='Run the source code tidiness check',
              category='testing')
     @CommandArgument('--faster', default=False, action="store_true",
-                     help="Only check changed files and skip the WPT lint in tidy")
+                     help="Only check changed files and skip the WPT lint in tidy, "
+                          "if no changes in the WPT files")
```
Comment (Wafflespeanut) [inline python/servo/testing_commands.py]: ... (if there are no changes ...)

**Comment by Wafflespeanut** [comment]
You're right. The help message is confusing. Can you squash your commits once you make the change I've suggested, and we'll land this?

**Comment by zakorgy** [comment]
Unfortunately the current code prints the following when './mach test-tidy' executed without faster:
  <pre>Progress: 0% (16/7812)
 Checking files for tidiness...
  Progress: 100% (7812/7812)
Running the WPT lint...
  Progress: 100% (18748/18748)
tidy reported no errors.</pre>

**Comment by Wafflespeanut** [comment]
Good catch! You're right that the progress is printed because `is_iter_empty` executes once. This can be fixed by changing the `\n` to `\r` [here](https://github.com/zakorgy/servo/blob/061a643a63ab98d6dfef7c7dca68185d8cc1babe/python/tidy.py#L572). Well, not exactly a fix, but a workaround for now. (btw, remove the space there, I dunno how it got there in the first place).

**Comment by Wafflespeanut** [comment]
Awesome! Thanks for doing this! :)

@bors-servo r+

**Pull request merged**